src/conn/conn_compact.c

```diff
@@ -115,6 +202,12 @@ __background_compact_should_run(WT_SESSION_IMPL *session, const char *uri, int64
 
     conn = S2C(session);
 
+    /* Check if the file is excluded. */
+    if (__background_compact_find_exclude_entry(session, uri)) {
```

Comment:
Here, I think we should not search every time. If we find that a table is excluded or not, we should keep this information so we don't need to call this function again.

Reply:
This seems okay to me, I'm not sure where else to store that. Storing it in the stats list will be the same complexity as this.

Reply:
It is not a performance path, and for the list size that can be given through the API, I don't think we need to optimise it prematurely. This should be okay.

Reply:
Thank you both, I will leave it as is for now.